public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -316,8 +316,17 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
         const timeRange = { startTime: options.range.from.unix(), endTime: options.range.to.unix() };
         const query = this.applyVariables(targets.nativeSearch[0], options.scopedVars);
         const searchQuery = this.buildSearchQuery(query, timeRange);
+        // Params can't have undefined values that are possible in SearchQueryParams
+        const params: Record<string, string | number> = {};
+        for (const key in searchQuery) {
+          const value = searchQuery[key as keyof SearchQueryParams];
```

Comment:
It is a good opportunity  to either fix this type or add a `//eslint-ignore-next-line` and remove this from the betterer list

Reply:
Good feedback! Added in https://github.com/grafana/grafana/pull/78468/commits/8ab76870693e47fadb047adcab897e929b69785c (plus I added it to other 2 places where `key as keyof` is used in the same file.